### PR TITLE
DPC-926: Updated Backend Services to POST form params

### PR DESCRIPTION
**Pull request by nickrobison-usds**
**Why**

One of the Google Group uses reported that they were unable to generate access tokens from the `Token/auth` endpoint. The issue was that their query params were being stripped and never making it to our endpoint. I'm not sure why their library wasn't working, but it revealed a poor implementation of the Backend Services spec where we should have been receiving the auth values as url encoded form params, in the post body, rather than as query params.

This will be a breaking change for our users, but hopefully a simple one.

**What Changed**

Updated authentication implementation to move from requiring query params to requiring encoded form params in the body. This actually matches the backend services spec, our previous implementation was incorrect.

All of our code and helper scripts have been updated to account for this. This WILL be a breaking change for users, but it should also fix some of the issues that have been reported.

**Choices Made**

**Tickets closed**:

DPC-926

**Future Work**

**Checklist**

- [x] All tests are passing via `make ci-app` (app change) and `make ci-web` (website change)
- [x] Swagger documentation has been updated
- [x] FHIR documentation has been updated
- [x] Any required dpc-ops changes have a PR submitted and mentioned in this ticket
- [x] Any manual migration steps are documented, scripts written (where applicable), and tested
- [x] Before merging, any required dpc-ops changes have been approved and merged into master of the dpc-ops repo

### CODE REVIEW

**Comment by nickrobison-usds** [comment]
Confirmed at the FHIR connectathon that this fix works correctly. We'll need to work out a messaging strategy to ensure we handle the breaking change appropriately.

File: dpc-testing/src/main/java/gov/cms/dpc/testing/APIAuthHelpers.java [hunk 1]
```diff
@@ -132,6 +144,21 @@ public static AuthResponse jwtAuthFlow(String baseURL, String macaroon, UUID key
         return authResponse;
     }
 
+    private static NameValuePair encodeNameValuePair(String name, String value) {
```
Comment (embh) [inline dpc-testing/src/main/java/gov/cms/dpc/testing/APIAuthHelpers.java]: Would `BasicNameValuePair` serve the same purpose? https://hc.apache.org/httpcomponents-core-4.4.x/httpcore/apidocs/org/apache/http/message/BasicNameValuePair.html

Comment (nickrobison-usds) [reply dpc-testing/src/main/java/gov/cms/dpc/testing/APIAuthHelpers.java]: Yes! Thank you! Not sure how I missed that the first time around.

File: dpc-testing/src/main/java/gov/cms/dpc/testing/APIAuthHelpers.java [hunk 2]
```diff
@@ -120,6 +132,7 @@ public static AuthResponse jwtAuthFlow(String baseURL, String macaroon, UUID key
             builder.addParameter("client_assertion_type", CLIENT_ASSERTION_TYPE);
             builder.addParameter("client_assertion", jwt);
```
Comment (embh) [inline dpc-testing/src/main/java/gov/cms/dpc/testing/APIAuthHelpers.java]: With the `HttpPost` entity being the form data, are parameters still needed on the URL?

File: dpc-api/src/test/java/gov/cms/dpc/api/auth/jwt/JWTUnitTests.java [hunk 3]
```diff
@@ -68,50 +70,51 @@ private JWTUnitTests() {
 
     @Nested
     @DisplayName("Query Param tests")
```
Comment (embh) [inline dpc-api/src/test/java/gov/cms/dpc/api/auth/jwt/JWTUnitTests.java]:
```suggestion
    @DisplayName("Form Param tests")
```

File: dpc-testing/src/main/java/gov/cms/dpc/testing/APIAuthHelpers.java (hunk 2 above)
Comment (nickrobison-usds) [reply dpc-testing/src/main/java/gov/cms/dpc/testing/APIAuthHelpers.java]: They are not! Removed.

File: dpc-api/src/test/java/gov/cms/dpc/api/auth/jwt/JWTUnitTests.java (hunk 3 above)
Comment (nickrobison-usds) [reply dpc-api/src/test/java/gov/cms/dpc/api/auth/jwt/JWTUnitTests.java]: Fixed!

**Pull request merged**